### PR TITLE
Allow custom error explainers

### PR DESCRIPTION
**Pull request by cpitclaudel**
This PR has two parts: cleaning up incorrect uses of call-process (typically, by calling it without consulting the checker executable variable), and adding support for URLs as requested in #1192 (cc @marsam)

### CODE REVIEW

**Comment by cpitclaudel** [comment]
@marsam good idea, I revised the docs to add an example of an `:error-explainer` and document other properties that we hadn't documented up to this point.

File: flycheck.el [hunk 1]
```diff
@@ -9547,7 +9587,12 @@ See URL `https://eslint.org/'."
        (flycheck-verification-result-new
         :label "config file"
         :message (if have-config "found" "missing or incorrect")
-        :face (if have-config 'success '(bold error)))))))
+        :face (if have-config 'success '(bold error))))))
+  :error-explainer
+  (lambda (err)
+    (let ((error-code (flycheck-error-id err))
+          (url "https://eslint.org/docs/rules/%s"))
+      (and error-code `(url . ,(format url error-code))))))
```
Comment (marsam) [inline flycheck.el:9595]: I think we should skip non-builtin rules:
```suggestion
      (unless (cl-find ?/ error-code) ; skip non-builtin rules
        `(url . ,(format url error-code))))))
```

File: flycheck.el [hunk 2]
```diff
@@ -10443,16 +10494,12 @@ expression, which selects linters for lintr."
...
-    (let ((process-environment (append '("LC_ALL=C") process-environment)))
-      (call-process R nil t nil
-                    "--slave" "--restore" "--no-save" "-e"
-                    "library('lintr')")
-      (goto-char (point-min))
-      (not (re-search-forward "there is no package called 'lintr'"
-                              nil 'no-error)))))
+(defun flycheck-r-has-lintr (checker)
+  "Whether CHECKER (R) has installed the `lintr' library."
+  (eq 0 (flycheck-call-checker-process
```
Comment (marsam) [inline flycheck.el:10499]: nitpick:
```suggestion
  (zerop (flycheck-call-checker-process
```

Comment (cpitclaudel) [reply flycheck.el:10499]: :thinking: I think that will raise an error if the return value is not a number, but call-process can return strings (and our wrapper can return nil, as well)

Comment (cpitclaudel) [reply flycheck.el:10499]: That being said, eql would be better than eq here. Changed.

File: flycheck.el (hunk 1 above)
Comment (cpitclaudel) [reply flycheck.el]: Good point, committed.

Comment (cpitclaudel) [reply flycheck.el]: I did keep the `error-code` test though, because we don't want to display a page when the error code is nil.

**Comment by cpitclaudel** [comment]
A plague on github's "commit suggestion" — why doesn't it credit the author properly?

**Pull request merged**